tests/data_interface/test_data_parameters.py

```diff
@@ -0,0 +1,514 @@
+"""
+
+from unittest.mock import MagicMock, Mock, patch
+
+import geopandas as gpd
+import pandas as pd
+import pytest
+from shapely.geometry import box
+
+from climakitae.core.data_interface import DataParameters
+
+
+def mock_scenario_to_experiment_id(scenario, reverse=False):
+    """Mock implementation of scenario_to_experiment_id with additional scenarios"""
+    scenario_dict = {
+        "Historical Reconstruction": "reanalysis",
+        "Historical Climate": "historical",
+        "SSP 2-4.5": "ssp245",
+        "SSP 5-8.5": "ssp585",
+        "SSP 3-7.0": "ssp370",
+        "SSP 1-2.6": "ssp126",  # Add the missing scenario
+    }
+
+    if reverse:
+        scenario_dict = {v: k for k, v in scenario_dict.items()}
+
+    # Handle special cases used in tests
+    if scenario == "n/a" and not reverse:
+        return "n/a"
+    if scenario == "n/a" and reverse:
+        return "n/a"
+
+    return scenario_dict.get(scenario, scenario)
+
+
+class TestDataParameters:
+    """
+    Tests for the DataParameters class and its methods.
+    """
+
+    @pytest.fixture
+    def mock_data_interface(self):
+        """Fixture to create a mocked DataInterface to avoid external dependencies."""
+        with patch("climakitae.core.data_interface.DataInterface") as mock_di:
+            # Create mock instance with required properties
+            mock_instance = Mock()
+
+            # Mock variable descriptions
+            var_desc = pd.DataFrame(
+                {
+                    "variable_id": [
+                        "tas",
+                        "pr",
+                        "tas_derived_index",
+                        "pr_derived_index",
+                    ],
+                    "display_name": [
+                        "Air Temperature at 2m",
+                        "Precipitation",
+                        "Heat Index",
+                        "Heavy Precipitation Index",
+                    ],
+                    "downscaling_method": [
+                        "Dynamical",
+                        "Dynamical",
+                        "Dynamical",
+                        "Dynamical",
+                    ],
+                    "timescale": [
+                        "hourly, daily, monthly",
+                        "daily, monthly",
+                        "hourly",
+                        "hourly",
+                    ],
+                    "unit": ["°C", "mm day⁻¹", "°C", "mm day⁻¹"],
+                    "colormap": ["viridis", "Blues", "hot", "Blues"],
+                    "extended_description": [
+                        "Temperature",
+                        "Rain",
+                        "Heat Index",
+                        "Heavy rain",
+                    ],
+                    "show": [True, True, True, True],
+                }
+            )
+            mock_instance.variable_descriptions = var_desc
```

Comment:
Why not just read in the csv file? Shouldn’t add too much overhead to the test time and then you could guarantee that all the info is correct for each variable.

Reply:
Agreed, in this instance